specutils/tests/test_loaders.py

```diff
@@ -321,10 +322,43 @@ def test_tabular_fits_writer(tmpdir, spectral_axis):
         spectrum.write(tmpfile, format='tabular-fits')
     spectrum.write(tmpfile, format='tabular-fits', overwrite=True)
 
-    table = Table.read(tmpfile)
-    assert table['flux'].unit == spectrum.flux.unit
+    cmap = {spectral_axis: ('spectral_axis', wlu[spectral_axis]),
+            'flux': ('flux', 'erg / (s cm**2 AA)'),
+            'uncertainty': ('uncertainty', None)}
 
-    # ToDo: get tabular_fits_loader to also read this in correctly!
+    # Read it back again and check against the original
+    spec = Spectrum1D.read(tmpfile, format='tabular-fits', column_mapping=cmap)
+    assert spec.flux.unit == u.Unit('erg / (s cm**2 AA)')
+    assert spec.spectral_axis.unit == spectrum.spectral_axis.unit
+    assert quantity_allclose(spec.spectral_axis, spectrum.spectral_axis)
+    assert quantity_allclose(spec.flux, spectrum.flux)
+    assert quantity_allclose(spec.uncertainty.quantity,
+                             spectrum.uncertainty.quantity)
+
+
+@pytest.mark.parametrize("spectral_axis",
+                         ['wavelength', 'frequency', 'energy', 'wavenumber'])
+def test_tabular_fits_2d(tmpdir, spectral_axis):
+    wlu = {'wavelength': u.AA, 'frequency': u.GHz, 'energy': u.eV,
+           'wavenumber': u.cm**-1}
+    # Create a small data set with 2D flux + uncertainty
+    disp = np.arange(1, 1.1, 0.01)*wlu[spectral_axis]
```

Comment:
Eventually we're going to want to ensure that any spectral axis in frequency space is strictly descending. I'm not sure we need to worry about that case right now, but something to think about.

```suggestion
    disp = np.arange(1, 1.1, 0.01)*wlu[spectral_axis]
    
    if spectral_axis == 'frequency':
        disp = disp[::-1]
```

Reply:
> Eventually we're going to want to ensure that any spectral axis in frequency space is strictly descending. I'm not sure we need to worry about that case right now, but something to think about.

That's actually everything but wavelength, right? Yes, if there is functionality that depends on descending order in frequency/energy, that's probably good to keep as reminder.